client/matching/client.go

```diff
@@ -232,6 +232,30 @@ func (c *clientImpl) ListTaskListPartitions(
 	return client.ListTaskListPartitions(ctx, request, opts...)
 }
 
+func (c *clientImpl) GetTaskListsByDomain(
+	ctx context.Context,
+	request *types.MatchingGetTaskListsByDomainRequest,
+	opts ...yarpc.CallOption,
+) (*types.GetTaskListsByDomainResponse, error) {
+	opts = common.AggregateYarpcOptions(ctx, opts...)
+
+	// This result could be large
+	clients := c.clients.GetAllClients()
```

Comment:
This only returns all cached matching clients on the hosts right? If a matching host is never cached on this host then it won't be returned and we may miss tasks list there.

Reply:
Change to use get memebers from ringpop